### PR TITLE
Fix/owner cleanup delegation

### PR DESCRIPTION
**Pull request by sasurobert**
- fixed the wrong owner key hex encoding
- fixed the getUnStakedTokensList view function

### CODE REVIEW

File: epochStart/metachain/stakingDataProvider.go [hunk 1]
```diff
@@ -256,10 +256,7 @@ func (sdp *stakingDataProvider) getValidatorDataFromStakingSC(validatorAddress s
 }
 
 func (sdp *stakingDataProvider) getValidatorInfoFromSC(validatorAddress string) (*big.Int, *big.Int, [][]byte, error) {
-	validatorAddressBytes, err := hex.DecodeString(validatorAddress)
-	if err != nil {
-		return nil, nil, nil, err
-	}
+	validatorAddressBytes := []byte(validatorAddress)
```
Comment (raduchis) [inline epochStart/metachain/stakingDataProvider.go:259]: is there no problem with the backwards compatibility?

File: vm/systemSmartContracts/delegation.go [hunk 2]
```diff
@@ -207,7 +207,7 @@ func (d *delegation) Execute(args *vmcommon.ContractCallInput) vmcommon.ReturnCo
 		return d.getUserUnStakedValue(args)
 	case "getUserUnBondable":
 		return d.getUserUnBondable(args)
-	case "getUserUnDelegatedList":
+	case "getUserUndelegatedList":
```
Comment (raduchis) [inline vm/systemSmartContracts/delegation.go:210]: was this change necessary? other methods are unDelegate, unBond, unStake, unJail

File: vm/systemSmartContracts/delegation.go [hunk 3]
```diff
@@ -839,6 +839,7 @@ func (d *delegation) unBondNodes(args *vmcommon.ContractCallInput) vmcommon.Retu
 	successKeys, _ := getSuccessAndUnSuccessKeys(vmOutput.ReturnData, args.Arguments)
 	for _, successKey := range successKeys {
 		status.UnStakedKeys, status.NotStakedKeys = moveNodeFromList(status.UnStakedKeys, status.NotStakedKeys, successKey)
+		status.StakedKeys, status.NotStakedKeys = moveNodeFromList(status.StakedKeys, status.NotStakedKeys, successKey)
```
Comment (raduchis) [inline vm/systemSmartContracts/delegation.go:842]: is there a case when a stakedKey moves directly to notStaked, without being unstaked first?

File: vm/systemSmartContracts/staking.go [hunk 4]
```diff
@@ -1447,7 +1447,7 @@ func (s *stakingSC) getOwner(args *vmcommon.ContractCallInput) vmcommon.ReturnCo
 		return vmcommon.UserError
 	}
 
-	s.eei.Finish([]byte(hex.EncodeToString(stakedData.OwnerAddress)))
+	s.eei.Finish(stakedData.OwnerAddress)
```
Comment (raduchis) [inline vm/systemSmartContracts/staking.go:1450]: There still are some s.eei.Finish() with hexEncode on the RewardAddress: lines 1273 and 1384. Should not all be not hexEncoded?

**Review by raduchis — CHANGES_REQUESTED** [review]

File: epochStart/metachain/stakingDataProvider.go (hunk 1 above)
Comment (raduchis) [reply epochStart/metachain/stakingDataProvider.go:259]: also please add description to the PR

**Review by SebastianMarian — APPROVED** [review]

File: vm/systemSmartContracts/staking.go (hunk 4 above)
Comment (iulianpascalau) [reply vm/systemSmartContracts/staking.go:1450]: I have implemented this and indeed I have used the other examples in the system SC. What we have figure out today is that this hex encoding is useless as we need a complementary hex.Decode call when processing internal data. The code is now slightly optimized and one bug was fixed. If we need an API function for the owner field we will create a new function that will return the data in hex format.

**Review by SebastianMarian — APPROVED** [review]

File: vm/systemSmartContracts/delegation.go (hunk 2 above)
Comment (raduchis) [reply vm/systemSmartContracts/delegation.go]: ok

File: epochStart/metachain/stakingDataProvider.go (hunk 1 above)
Comment (raduchis) [reply epochStart/metachain/stakingDataProvider.go:259]: ok

File: vm/systemSmartContracts/delegation.go (hunk 3 above)
Comment (sasurobert) [reply vm/systemSmartContracts/delegation.go:842]: If there is a desynchronization error between the contracts. Only safety measures to not block states.

File: integrationTests/vm/delegation/delegationScenarios_test.go [hunk 5]
```diff
@@ -70,27 +70,32 @@ func TestDelegationSystemNodesOperations(t *testing.T) {
 	checkNodesStatus(t, tpn, vm.StakingSCAddress, blsKeys[:numNodesToStake], "staked")
 
 	//unStake 3 nodes
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply integrationTests/vm/delegation/delegationScenarios_test.go:72]: Not more relevant

**Review by sasurobert — COMMENTED** [review]
Green from me

File: vm/systemSmartContracts/delegation.go (hunk 3 above)
Comment (raduchis) [reply vm/systemSmartContracts/delegation.go:842]: ok

**Review by raduchis — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by LucianMincu — APPROVED** [review]
